docs/02_types/interfaces.md

```diff
@@ -5,4 +5,194 @@ path: /docs/types/interfaces
+
+```graphql
+query {
+  customers {
+    name
+  }
+}
+```
+
+Whether the object is a `Company` or an `Individual`, it doesn’t matter – you
+still get their name. If you want some object-specific fields, you can query
+them with an inline fragment, for example:
+
+```graphql
+query {
+  customers {
+    name
+    ... on Individual {
+      company {
+        name
+      }
+    }
+  }
+}
+```
+
+Interfaces are a good choice whenever you have a set of objects that are used
+interchangeably, and they have several significant fields in common. When they
+don’t have fields in common, use a [Union](/docs/types/unions) instead.
+
+## Defining interfaces
+
+Interfaces are defined using the `@strawberry.interface` decorator:
+
+```python+schema
+import strawberry
+
+@strawberry.interface
+class Customer:
+    name: str
+---
+interface Customer {
+  name: String!
+}
+```
+
+> Note: Interface classes should never be instantiated directly.
+
+## Implementing interfaces
+
+To define object type that implement an interface the type must inherit from the
+interface:
+
+```python
+import strawberry
+
+@strawberry.type
+class Individual(Customer):
+    # additional fields
+
+@strawberry.type
+class Company(Customer):
+    # additional fields
+```
+
+> ⚠️  If you add an object type which implements an interface, but that object
+> type doesn’t appear in your schema as a field return type or a union member,
+> then you will need to add that object to the Schema defintion directly.
+
+Interfaces can also implement other interfaces:
+
+```python+schema
+import strawberry
+
+@strawberry.interface
+class Error:
+    message: str
+
+@strawberry.interface
+class FieldError(Error):
+    message: str
+    field: str
+
+@strawberry.type
+class PasswordTooShort(FieldError):
+    message: str
+    field: str
+    fix: str
+---
+interface Error {
+  message: String!
+}
+
+interface FieldError implements Error {
+  message: String!
+  field: String!
+}
+
+type PasswordTooShort implements FieldError & Error {
+  message: String!
+  field: String!
+  minLength: Int!
+}
+```
+
+## Implementing fields
+
+Interfaces can provide field implementations as well. For example:
+
+```python
+import strawberry
+
+@strawberry.interface
+class Customer:
+    @strawberry.field
+    def name(self) -> str:
+        return self.name.title()
+```
+
+This resolve method will be called by objects who implement the interface.
+Object classes can override the implmentation by defining their own `name`
+field:
+
+```python
+import strawberry
+
+@strawberry.type
+class Company(Customer):
+    @strawberry.field
+    def name(self) -> str:
+        return f"{self.name} Limited"
+```
+
+
+## Resolving an interface
+
+When a field’s return type is an interface, GraphQL needs to know what specific
+object type to use for the return value. In the example above, each customer
+must be categorized as an `Individual` or `Company`. To do this you need to
+always return an instance of an object type:
```

Comment:
The example for this is kinda strange, as you're not returning "an instance", but instead a _list_ of instances, so I'm not exactly sure what you're trying to describe. Maybe add an example of what _not_ to do?

Reply:
Hmm yeah I'm trying to get across that you can't return a plain object or an instance of the interface because then GraphQL won't know what object type you're referring to. I agree it's confusing that in the example I'm returning a list but it's irrelevant if it's a list or not. I'll change the example to just return a single value so it's not as confusing.

Reply:
@BryceBeagle updated the example. Does that help?